src/Notification_NotificationTemplate.php

```diff
@@ -500,10 +500,7 @@ public static function getModeClass($mode, $extratype = '')
         } else if ($extratype === 'setting') {
             $classname = 'Notification' . ucfirst($mode) . 'Setting';
         } else {
-            if ($extratype !== '') {
-                throw new \LogicException(sprintf('Unknown type `%s`.', $extratype));
-            }
-            $classname = 'Notification' . ucfirst($mode);
+            throw new \LogicException(sprintf('Unknown type `%s`.', $extratype));
         }
```

Comment:
This change is not correct. `Notification_NotificationTemplate::getModeClass('mailing')` is a valid call, and the expected returned class is `NotificationMailing`.